File: haskell/day01.hs
main :: IO () 
main = do  
    f <- readFile "../input.txt" 
    let depths = map read $ lines f :: [Int]  
    print $ solve depths 1 -- Part One 
    print $ solve depths 3 -- Part Two 
    
solve :: Ord a => [a] -> Int -> Int
solve depths window = sum $ fromEnum <$> zipWith (<) depths (drop window depths)

File: day01/python/day01.py
def solve(depths, window):
    return sum(x < y for (x, y) in zip(depths, depths[window:]))

with open("../input.txt")as file:
    depths = [int(n) for n in file.readlines()] 
    print(solve(depths, 1)); # Part One 
    print(solve(depths, 3)); # Part Two 
